gitpod-io/gitpod
# [ws-proxy] Serve port-not-found page when port isn't answering

This PR fixes the "port not responding" page.

In the past we'd proxyPass to the dashboard. That was possible because `proxy` did not implemented the meta/workspace split. ws-proxy does, and clearly lives on the workspace side, hence we don't want to just proxy pass to dashboard, as we might not have a dashboard available. Also the "port-not-found" page has absolutely no dashboard dependencies and just lived there for convenience's sake.

Instead of proxying to the dashboard, we move the "port-not-found.html" page to ws-proxy. This means ws-proxy can now serve built-in pages, something that might come in handy for other error situations, too.

### How to test
1. Start a workspace http://cw-fix-2170.staging.gitpod-dev.com/#https://github.com/gitpod-io/django-locallibrary-tutorial
2. Open a non-existent/non-exposed port URL in your browser, e.g. `1010-<workspace-id>.cw-fix-2170.staging.gitpod-dev.com` (note: you cannot use `gp url` here, because of #2212)

You should see the usual "port not responding" page after about 30 seconds.
